sdk/core/Azure.Core/src/Shared/DiagnosticScope.cs

```diff
@@ -136,6 +137,9 @@ public void Dispose()
         /// Marks the scope as failed.
         /// </summary>
         /// <param name="exception">The exception to associate with the failed scope.</param>
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026", Justification = "The Exception being passed into this method has public properties preserved on the inner method MarkFailed." +
+            "The public property System.Exception.TargetSite.get is not compatible with trimming and produces a warning when preserving all public properties. Since we do not use this property, and" +
+            "use of this property by Application Insights cannot be supported with trimming, we can suppress the warning coming from the inner method.")]
```

Comment:
Technically, the fact that AppInsight can't use the property when trimmed is actually a very good reason to keep the warning. I think the justification is that we don't think AppInsight will ever try to access that property. If that's not true, then we would need to do more here (either change AppInsight or try to find a different solution).

Reply:
Got it, I'll tweak the justification. As far as I'm aware, AppInsights doesn't use this property, so we should be ok to suppress it.